### ISSUE TITLE
MapChart not rendered with Angular

### ISSUE THREAD

**Issue by samuel-p** [amcharts/amcharts4#296]
I tried to implement the example from https://codepen.io/team/amcharts/pen/yvdwrR with Angular, but the map is not rendered and there is no error message.

Here is my config:
```
import {AfterViewInit, Component, ElementRef, NgZone, OnDestroy, ViewChild} from '@angular/core';
import * as am4core from "@amcharts/amcharts4/core";
import * as am4maps from "@amcharts/amcharts4/maps";
import am4geodata_worldLow from "@amcharts/amcharts4-geodata/worldLow";

@Component({
  selector: 'app-chart-example',
  templateUrl: './chart-example.component.html'
})
export class ChartExampleComponent implements AfterViewInit, OnDestroy {
  @ViewChild('chart') div: ElementRef;
  private chart: am4maps.MapChart;

  constructor(private zone: NgZone) {}

  ngAfterViewInit(): void {
    this.zone.runOutsideAngular(() => {
      var chart = am4core.create(this.div.nativeElement, am4maps.MapChart);
      chart.geodata = am4geodata_worldLow;
      chart.projection = new am4maps.projections.Miller();

      var title = chart.chartContainer.createChild(am4core.Label);
      title.text = "Life expectancy in the World";
      title.fontSize = 20;
      title.paddingTop = 30;
      title.align = "center";

      var polygonSeries = chart.series.push(new am4maps.MapPolygonSeries());
      var polygonTemplate = polygonSeries.mapPolygons.template;
      polygonTemplate.tooltipText = "{name}: {value.value.formatNumber('#.0')}";
      polygonSeries.heatRules.push({
        property: "fill",
        target: polygonSeries.mapPolygons.template,
        min: am4core.color("#ffffff"),
        max: am4core.color("#AAAA00")
      });
      polygonSeries.useGeodata = true;

      // add heat legend
      var heatLegend = chart.chartContainer.createChild(am4maps.HeatLegend);
      heatLegend.valign = "bottom";
      heatLegend.align = "left";
      heatLegend.width = am4core.percent(100);
      heatLegend.series = polygonSeries;
      heatLegend.orientation = "horizontal";
      heatLegend.margin(20, 20, 20, 20);
      heatLegend.valueAxis.renderer.labels.template.fontSize = 10;
      heatLegend.valueAxis.renderer.minGridDistance = 40;

      polygonSeries.mapPolygons.template.events.on("over", event => {
        handleHover(event.target);
      });

      polygonSeries.mapPolygons.template.events.on("hit", event => {
        handleHover(event.target);
      });

      function handleHover(mapPolygon) {
        if (!isNaN(mapPolygon.dataItem.value)) {
          heatLegend.valueAxis.showTooltipAt(mapPolygon.dataItem.value);
        } else {
          heatLegend.valueAxis.hideTooltip();
        }
      }

      polygonSeries.mapPolygons.template.strokeOpacity = 0.4;
      polygonSeries.mapPolygons.template.events.on("out", event => {
        heatLegend.valueAxis.hideTooltip();
      });

      chart.zoomControl = new am4maps.ZoomControl();

      // life expectancy data
      polygonSeries.data = [
        {id: "AF", value: 60.524},
        ... ,
        {id: "ZW", value: 58.142}
      ];


      // excludes Antarctica
      polygonSeries.exclude = ["AQ"];

      this.chart = chart;
    });
  }

  ngOnDestroy(): void {
    this.zone.runOutsideAngular(() => {
      if (this.chart) {
        this.chart.dispose();
      }
    });
  }
}
```

Here is a screensot.
![screenshot](https://user-images.githubusercontent.com/7959516/44721158-481feb80-aac9-11e8-8a57-318270fa1f1e.png)

The title and the legend are rendered, but the map is missing. I tested Firefox, Opera and Chromium with no differences.

**Comment by RobbiesCode** [comment]
Hi,

I have tried reproducing your issue (with the exact same setup) but the map shows up just like it should.

I have only replaced

```
var chart = am4core.create(this.div.nativeElement, am4maps.MapChart);
```

With

```
var chart = am4core.create("chartdiv", am4maps.MapChart);
```

As I did not know what `this.div.nativeElement` referred to.

In your component's HTML file, make sure your chartdiv has a width & height set:

```
<div id="chartdiv" style="width: 500px; height: 500px;"></div>
```

Also make sure the amCharts 4 geodata is installed:

```
npm install @amcharts/amcharts4-geodata
```

All these things combined should make the map show up. At least, it does for me. Please let us know if the issue persists.

**Comment by samuel-p** [comment]
Thanks for your reply. I tested my application again and noticed that it does not render if the element is hidden at the beginning, like in the following example:

chart-example.component.html
```
<button (click)="hide = !hide">{{ hide ? 'Show' : 'Hide' }}</button>
<div [hidden]="hide">
  <div id="chartdiv" style="background-color: white; width: 500px; height: 500px;"></div>
</div>
```

chart-example.component.ts
```
@Component({
  selector: 'app-chart-example',
  templateUrl: './chart-example.component.html'
})
export class ChartExampleComponent implements AfterViewInit, OnDestroy {
  private chart: am4maps.MapChart;
  hide = true;

  constructor(private zone: NgZone) {
  }

  ngAfterViewInit(): void {
    this.zone.runOutsideAngular(() => {
      var chart = am4core.create("chartdiv", am4maps.MapChart);
      chart.geodata = am4geodata_worldLow;
      chart.projection = new am4maps.projections.Miller();

      var title = chart.chartContainer.createChild(am4core.Label);
      title.text = "Life expectancy in the World";
      title.fontSize = 20;
      title.paddingTop = 30;
      title.align = "center";

      var polygonSeries = chart.series.push(new am4maps.MapPolygonSeries());
      var polygonTemplate = polygonSeries.mapPolygons.template;
      polygonTemplate.tooltipText = "{name}: {value.value.formatNumber('#.0')}";
      polygonSeries.heatRules.push({
        property: "fill",
        target: polygonSeries.mapPolygons.template,
        min: am4core.color("#ffffff"),
        max: am4core.color("#AAAA00")
      });
      polygonSeries.useGeodata = true;


      // add heat legend
      var heatLegend = chart.chartContainer.createChild(am4maps.HeatLegend);
      heatLegend.valign = "bottom";
      heatLegend.align = "left";
      heatLegend.width = am4core.percent(100);
      heatLegend.series = polygonSeries;
      heatLegend.orientation = "horizontal";
      heatLegend.margin(20, 20, 20, 20);
      heatLegend.valueAxis.renderer.labels.template.fontSize = 10;
      heatLegend.valueAxis.renderer.minGridDistance = 40;

      polygonSeries.mapPolygons.template.events.on("over", event => {
        handleHover(event.target);
      });

      polygonSeries.mapPolygons.template.events.on("hit", event => {
        handleHover(event.target);
      });

      function handleHover(mapPolygon) {
        if (!isNaN(mapPolygon.dataItem.value)) {
          heatLegend.valueAxis.showTooltipAt(mapPolygon.dataItem.value);
        } else {
          heatLegend.valueAxis.hideTooltip();
        }
      }

      polygonSeries.mapPolygons.template.strokeOpacity = 0.4;
      polygonSeries.mapPolygons.template.events.on("out", event => {
        heatLegend.valueAxis.hideTooltip();
      });

      chart.zoomControl = new am4maps.ZoomControl();

      // life expectancy data
      polygonSeries.data = [
        {id: "AF", value: 60.524},
        ... ,
        {id: "ZW", value: 58.142}
      ];

      // excludes Antarctica
      polygonSeries.exclude = ["AQ"];

      this.chart = chart;
    });
  }

  ngOnDestroy(): void {
    this.zone.runOutsideAngular(() => {
      if (this.chart) {
        this.chart.dispose();
      }
    });
  }
}
```

If you set the default value for hide on false the chart renders perfectly. In v3 is renders perfectly in both cases.

**Comment by Pauan** [comment]
Codepen which reproduces the issue:

https://codepen.io/team/amcharts/pen/4abf3546db07f62ee1f2ac0ca5c74f7c

It only happens with map charts, not the other chart types.

**Comment by Pauan** [comment]
@zeroin Could you take a look at this?

**Comment by martynasma** [comment]
This seems to be related to #311.

**Comment by martynasma** [comment]
Fixed in [4.0.0-beta.50](https://github.com/amcharts/amcharts4/releases/tag/4.0.0-beta.50).

## [4.0.0-beta.50] - 2018-09-10

### Added
- `am4core.unuseAllThemes()` method added. Any chart created after this method call will not have any theme applied to it.
- New property `am4core.registry.baseSprites` will now hold all active (non-disposed) instances of top-level elements, such as charts.
- `startLocation`/`endLocation` properties added to `CategoryAxis` (previously available on `DateAxis` only). Allows to specify location the first and last cells of the axis should start on respectively.
- `innerRadius` of a Slice can now be set as `Percent` (e.g. `am4core.percent(50)`).
- You can now set custom radii (`radius` property) on Pie series (if you have more than one, you might want one to be larger and another smaller).

### Changed
- [`MapLine.imagesToConnect`](https://www.amcharts.com/docs/v4/reference/mapline/#imagesToConnect_property) now accepts an array of string IDs that can reference images by their `id` property.
- Changed `tooltip` inheritance so that it checks for `virtualParent` rather than direct `parent`. E.g. series' bullets use series' tooltip and not chart tooltip as it was before.

### Fixed
- Map polygon stroke thickness could be rendered incorrectly when zooming quickly. (Issue #175)
- Calling `addData()` more than once before data was validated resulted only last data item to be added. (Issue #222)
- `DateAxis.baseInterval` was ignored if data was set directly on series.
- `MapLine` was not paying attention to the properties set on template.
- `HeatLegend` was not updating its label count after its size changed. This could have been resulting in very few labels if initial container (div) size was small.
- The map portion of `MapChart` was incorrectly positioned after container (div) resize.
- Object properties, like `"name"` in GeoJSON were overriding same properties in `MapSeries` data. Now, the values specified in data are used over ones in GeoJSON. (Issue #307)
- Axis ranges used to reset `axisRange.label.text`. (Issue #308)
- Fixed issues with disposing series and charts.
- Updated to [canvg 1.5.3](https://github.com/canvg/canvg/releases/tag/v1.5.3) to fix em/rem text issue for epoxrt fallbacks.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by samuel-p** [comment]
It works perfectly now. Thank you.
